src/org/traccar/database/DeviceManager.java

```diff
@@ -420,4 +427,46 @@ public void resetTotalDistance(DeviceTotalDistance deviceTotalDistance) throws S
             throw new IllegalArgumentException();
         }
     }
+
+    public void sendCommand(Command command) throws RecoverablePduException, UnrecoverablePduException,
+            SmppTimeoutException, SmppChannelException, InterruptedException {
+        if (command.getSms()) {
```

Comment:
Should we automatically try SMS command if device is not online? Possibly a configuration option.